src/utils.ts

```diff
@@ -0,0 +1,16 @@
+export function parseArgs<O extends object>(
+  args: [O] | Array<O[keyof O]>,
+  propNames: Array<keyof O>
+): O {
+  const isOptionsObj = args.length === 1 && typeof args[0] === "object";
+
+  if (isOptionsObj) return args[0] as O;
+
+  return propNames.reduce((acc: O, p, index) => {
+    acc[p] = isOptionsObj
+      ? (args[0] as O)[p]
```

Comment:
`isOptionsObj` will always be false by the time we get here.

Reply:
Thank you for the review.
Unfortunately, I have had to focus on other projects at the moment, so I don't have the opportunity to make changes to the PR, which I will close.